### ISSUE TITLE
Nim for Ruby Programmers wiki would be helpful

### ISSUE THREAD

**Issue by BigBoneDaddy** [nim-lang/Nim#3008]
I think that a Nim for Ruby Programmers wiki would be useful. I don't know Ruby myself so I can't add it but I know some Rubyers who maybe would like Nim but they don't know how to start. They think it is easier to start Go or Rust so they maybe do them instead but I think that if Nim was easy for them to start then they would go Nim instead maybe.

**Comment by fowlmouth** [comment]
Don't you think singling out Ruby programmers would be discriminatory?

**Comment by TheRabbitologist** [comment]
Disclaimer: I'm not involved in the Nim project at all, and probably shouldn't even be commenting here. But... here's a question. Do you really think there would be a large enough potential draw to Nim from Ruby to warrant the creation of a guide? Nim for Python programmers makes sense, as Nim is rather Python-esque, and there definitely is a potential draw there, with that guide serving as a way to ease the transition. Nim for C programmers also makes sense, given that Nim compiles to C and is generally presented as a "systems-level" (my terminology) language, making Nim potentially useful for C programmers who want to reduce development time. I do not, however, see much of a potential draw from Ruby, speaking as someone who does some development work in Ruby. You said that you know some Ruby devs who would like to learn, but do you honestly think that interest is particularly common?

**Comment by reactormonk** [comment]
I started out as a Ruby developer and came to nim, but that was long ago. I probably know nim better than Ruby by now.

**Comment by BigBoneDaddy** [comment]
@TheRabbitologist There are many more Ruby dev than Python dev. Python never have extremely popular tech like Ruby on Rails. Ruby on Rails make there be lot of Ruby dev. Ruby has sheer numbers.

**Comment by kirbyfan64** [comment]
@BigBoneDaddy Yeah, except for Google, YouTube (technically still Google), Dropbox, Yahoo! Groups, reddit, NASA, Disney, and all those other tiny little companies. :)

Really, though, Ruby is *way* under Python on the TIOBE index. Python was created in the late 80s; Ruby came later.

**Comment by BitPuffin** [comment]
Large companies using them doesn't necessarily mean large userbase though. But sure, Python is more well established. But I'd say a lot of users of Python are unlikely to move because they are using Python for specific reasons. For instance, Python is very *very* popular in Animation (probably why Disney uses it), so many 3D Programs support Python scripting. Having a nim guide for those folks would be pointless because their 3D programs are still gonna support Python scripting and not Nim. I suspect that most Python users are gonna stay loyal to Python for various reasons. Ruby devs... seem more flexible.

**Comment by reactormonk** [comment]
And Ruby people usually use Ruby for RoR (I didn't), and nim doesn't have anything to close to RoR.

**Comment by BitPuffin** [comment]
Yeah, so until it does, maybe you should hold off on that guide thing. Make a nice web framework instead.

**Comment by Varriount** [comment]
Or just general tutorials/guides

**Comment by Varriount** [comment]
After some though, instead of closing this, I'll simply edit the title to reflect a broader desire.
